operator/src/main/java/org/keycloak/operator/Config.java

```diff
@@ -24,7 +24,7 @@
 /**
  * @author Vaclav Muzikar <vmuzikar@redhat.com>
  */
-@ConfigMapping(prefix = "operator")
+@ConfigMapping(prefix = "kc-operator")
 public interface Config {
     Keycloak keycloak();
```

Comment:
@shawkins 
What is the purpose of having the inner Keycloak interface included in the parent Config interface? Do we expect some other config props unrelated to Keycloak in this case? 

The thing is that when we put the `kc-` prefix to the config mapping, we explicitly reference the Keycloak Operator. What about merging these interfaces and thus simplify the configuration?

The breaking changes are inevitable anyway, right?

So, instead of:
```
-Doperator.keycloak.pod-labels."xxx"=
```

We would have:
```
-Dkc-operator.pod-labels."xxx"=
```

But that's my initial thought and I'm not aware of any issues it could bring, or violates certain k8s conventions, etc.

Reply:
In managed kafka for example we ended up with two deployments in our bundle - the operator and a sync component. Presumably the usage of the keycloak interface will allow for this possiblity as well:

kc-operator.sidecar.key=value

Reply:
> Presumably the usage of the keycloak interface will allow for this possiblity as well

+1. I think it worths the discussion about it as we're introducing breaking changes anyway, so we could polish it a little bit. @shawkins @vmuzikar WDYT?

Reply:
As @shawkins mentioned, the inner `Keycloak` interface is meant just for the Keycloak deployment just in case we have more deployments in the future.